packages/relay/src/lib/services/debugService/index.ts

```diff
@@ -0,0 +1,219 @@
+import type { Logger } from 'pino';
+import type { MirrorNodeClient } from '../../clients';
+import type { IDebugService } from './IDebugService';
+import type { CommonService } from '../ethService/ethCommonService';
+import { decodeErrorMessage, numberTo0x } from '../../../formatters';
+import constants from '../../constants';
+import { TracerType } from '../../constants';
+import { predefined } from '../../errors/JsonRpcError';
+
+export class DebugService implements IDebugService {
+  /**
+   * The interface through which we interact with the mirror node
+   * @private
+   */
+  private readonly mirrorNodeClient: MirrorNodeClient;
+
+  /**
+   * The logger used for logging all output from this class.
+   * @private
+   */
+  private readonly logger: Logger;
+
+  private readonly common: CommonService;
+
+  public readonly debugTraceTransaction = 'debug_traceTransaction';
+
+  constructor(mirrorNodeClient: MirrorNodeClient, logger: Logger, common: CommonService) {
+    this.logger = logger;
+    this.common = common;
+    this.mirrorNodeClient = mirrorNodeClient;
+  }
+
+  /**
+   * Checks if the Debug API is enabled
+   * @public
+   */
+  public static readonly isDebugAPIEnabled = process.env.DEBUG_API_ENABLED;
+
+  static requireDebugAPIEnabled(): void {
+    if (!process.env.DEBUG_API_ENABLED || process.env.DEBUG_API_ENABLED !== 'true') {
```

Comment:
Add this new env. variable in the `localAcceptance.env`, readme, docs and helm charts.